core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt

```diff
@@ -15,31 +15,28 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.main;
+package org.kiwix.kiwixmobile.core.main
 
-import android.content.Context;
-import java.io.BufferedReader;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-
-public class FileReader {
-
-  public String readFile(String filePath, Context context) {
-    try {
-      StringBuilder buf = new StringBuilder();
-      InputStream json = context.getAssets().open(filePath);
-      BufferedReader in =
-        new BufferedReader(new InputStreamReader(json, "UTF-8"));
-      String str;
-
-      while ((str = in.readLine()) != null) {
-        buf.append(str);
+import android.content.Context
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+class FileReader {
+  fun readFile(filePath: String?, context: Context): String {
+    return try {
```

Comment:
this can get drastically shorter with the power of kotlin.

InputStream has an extension method `bufferedReader` and from there you should investigate the methods on BufferedReader that can avoid this while loop all together